components/tidb_query_datatype/src/codec/mysql/json/binary.rs

```diff
@@ -88,6 +88,13 @@ impl<'a> JsonRef<'a> {
                     &self.value()[val_offset..val_offset + opaque_bytes_len as usize + len_len + 1],
                 )
             }
+            JsonType::Date | JsonType::Datetime | JsonType::Timestamp => {
```

Comment:
Is there any test case against these changes (including others in the `mysql/json` directory)

Reply:
Oops. It seems that the whole `get_entry` and `get_elem` doesn't have unit test. I'll add some test for them.

Reply:
Added!